java/src/main/native/src/TableJni.cpp

```diff
@@ -3446,11 +3449,30 @@ JNIEXPORT jobjectArray JNICALL Java_ai_rapids_cudf_Table_contiguousSplitGroups(
     // Resolves the split indices from offsets vector directly to avoid copying. Since
     // the offsets vector may be very large if there are too many small groups.
     std::vector<cudf::size_type> &split_indices = groups.offsets;
-    // Offsets laysout is [0, split indices..., num_rows] or [0] for empty keys, so
-    // need to removes the first and last elements.
-    split_indices.erase(split_indices.begin());
+    // Offsets layout is [0, split indices..., num_rows] or [0] for empty keys, so
+    // need to removes the first and last elements. First remove last one.
+    split_indices.pop_back();
+
+    // generate uniq keys by using `gather` method, this means remove the duplicated keys
+    std::unique_ptr<cudf::table> group_by_result_table = nullptr;
```

Comment:
NIT:
```suggestion
    std::unique_ptr<cudf::table> group_by_result_table;
```
Since `group_by_result_table ` is a local object, not a real pointer.

Reply:
Done